app/src/main/java/com/hedvig/app/feature/swedishbankid/sign/SwedishBankIdSignDialog.kt

```diff
@@ -58,61 +56,45 @@ class SwedishBankIdSignDialog : DialogFragment() {
         savedInstanceState: Bundle?,
     ) = ComposeView(requireContext()).apply {
         setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnLifecycleDestroyed(viewLifecycleOwner))
-        setContent {
-            val viewState by model.viewState.collectAsState()
-            HedvigTheme {
-                SwedishBankIdSignDialog(
-                    text = stringResource(getTextByViewState(viewState))
-                )
-            }
-        }
-    }
-
-    private fun getTextByViewState(viewState: SwedishBankIdSignViewModel.ViewState) = when (viewState) {
-        is SwedishBankIdSignViewModel.ViewState.StartClient -> R.string.SIGN_START_BANKID
-        SwedishBankIdSignViewModel.ViewState.Cancelled -> R.string.SIGN_CANCELED
-        SwedishBankIdSignViewModel.ViewState.InProgress -> R.string.SIGN_IN_PROGRESS
-        SwedishBankIdSignViewModel.ViewState.Success -> R.string.SIGN_SUCCESSFUL
-        SwedishBankIdSignViewModel.ViewState.Error -> R.string.SIGN_FAILED_REASON_UNKNOWN
-    }
 
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        model
-            .events
-            .flowWithLifecycle(viewLifecycle)
-            .onEach { event ->
-                when (event) {
-                    SwedishBankIdSignViewModel.Event.StartBankID -> {
+        var viewState: BankIdSignViewState by mutableStateOf(viewModel.viewState.value)
+        viewLifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.viewState.collect { state ->
```

Comment:
This preserves the behavior of only querying the CheckoutStatus and reacting to the changes while the app is on the foreground.